snap/channel.go

```diff
@@ -61,36 +64,51 @@ func ParseChannel(s string, architecture string) (Channel, error) {
 			risk = p[0]
 		} else {
 			track = p[0]
-			risk = "stable"
 		}
 	}
 
-	if !strutil.ListContains(channelRisks, risk) {
+	if risk != "" && !strutil.ListContains(channelRisks, risk) {
```

Comment:
this now parses foo//hot-fix  which I don't think we want,  and needs to be a new test

Reply:
By the looks of it, there's also a case of `/stable/` that is currently passing on master branch too.

Reply:
Cleaned that up in #6437, please take a look.